tests/src/test/scala/quasiquotes/ErrorSuite.scala

```diff
@@ -345,4 +365,102 @@ class ErrorSuite extends FunSuite {
+    assert(typecheckError("""
+      import scala.meta._
+      import scala.meta.dialects.Scala211
+      val p"$ref[..$tpes](..$apats)" = p"x[A, B]"
+    """) === """
+      |<macro>:4: pattern must be a value
+      |      val p"$ref[..$tpes](..$apats)" = p"x[A, B]"
+      |                                                ^
+    """.trim.stripMargin)
+  }
+
+  test("""p"$pat: $ptpe"""") {
+    assert(typecheckError("""
+      import scala.meta._
+      import scala.meta.dialects.Scala211
+      val pat = p"`x`"
+      val ptpe = pt"Y"
+      p"$pat: $ptpe"
+    """) === """
+      |<macro>:6: can't unquote a name here, use a pattern instead
+      |      p"$pat: $ptpe"
+      |        ^
+    """.trim.stripMargin)
+  }
+
+  test("p\"case $X: T =>\"") {
+    assert(typecheckError("""
+      import scala.meta._
+      import scala.meta.dialects.Scala211
+      val p"case $X: T => " = p"case x: T =>"
+    """) === """
+      |<macro>:4: not found: value X
+      |      val p"case $X: T => " = p"case x: T =>"
+      |                  ^
+    """.trim.stripMargin)
+  }
+
+//  test("""pt"$ptpe[..$ptpes]""") { // TODO review after #216 resolved
+//    assert(typecheckError("""
+//      import scala.meta._
+//      import scala.meta.dialects.Scala211
+//      val pt"$ptpe[..$ptpes]" = pt"x[y, z]"
+//    """).contains("found that tpe.isInstanceOf[Pat.Var.Type].`unary_!` is true"))
+//  }
+
+//  test("""pt"..$ptpes { ..$stats }"""") { // TODO review after #216 resolved
+//    assert(typecheckError("""
+//      import scala.meta._
+//      import scala.meta.dialects.Scala211
+//      val pt"..$ptpes { ..$stats }" = pt"x with y { val a: A; val b: B }"
+//    """).contains("found that tpes.forall(((tpe: scala.meta.internal.ast.Pat.Type) => tpe.isInstanceOf[Pat.Var.Type].`unary_!`.&&(tpe.isInstanceOf[Pat.Type.Wildcard].`unary_!`))) is false"))
+//  }
+
+//  test("""pt"$ptpe forSome { ..$stats }"""") { // TODO review after #216 resolved
+//    assert(typecheckError("""
+//      import scala.meta._
+//      import scala.meta.dialects.Scala211
+//      val pt"$ptpe forSome { ..$stats }" = pt"x forSome { val a: A }"
+//    """).contains("found that tpe.isInstanceOf[Pat.Var.Type].`unary_!` is true"))
+//  }
+
+//  test("""pt"$ptpe ..@$annots"""") { // TODO review after #216 resolved
+//    assert(typecheckError("""
+//      import scala.meta._
+//      import scala.meta.dialects.Scala211
+//      val pt"$ptpe ..@$annots" = pt"x @q @w"
+//    """).contains("found that tpe.isInstanceOf[Pat.Var.Type].`unary_!` is true"))
+//  }
+
+  test("""q"..$mods def this(...$paramss) = $expr"""") {
+    assert(typecheckError("""
+      import scala.meta._
+      import scala.meta.dialects.Scala211
+      q"private final def this(x: X, y: Y) = foo"
+    """) === """
+      |<macro>:4: this expected but identifier found
+      |      q"private final def this(x: X, y: Y) = foo"
+      |                                             ^
+    """.trim.stripMargin)
+  }
```

Comment:
Is there any particular order for tests in QuasiquoteSuite.scala? Would it be possible to order everything according to its order of appearance in quasiquote.md?